dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facilities.repo.ts

```diff
@@ -33,6 +34,48 @@ export class TfmFacilitiesRepo {
     return await collection.find({ 'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) } }).toArray();
   }
 
+  /**
+   * Finds the portal amendments across all facilities for a deal for a given status or set of statuses
+   * @param dealId - The deal id
+   * @param statuses - An array of portal amendment statuses to filter on
+   * @returns The matching portal amendments
+   */
+  public static async findPortalAmendmentsByDealIdAndStatus({
+    dealId,
+    statuses,
+  }: {
+    dealId: string | ObjectId;
+    statuses?: PortalAmendmentStatus[];
+  }): Promise<PortalFacilityAmendment[]> {
+    const collection = await this.getCollection();
+
+    const facilitiesOnDealWithPortalAmendments = await collection
+      .find(
+        {
+          'facilitySnapshot.dealId': { $eq: new ObjectId(dealId) },
+          'amendments.type': {
+            $eq: AMENDMENT_TYPES.PORTAL,
+          },
+        },
+        { projection: { amendments: 1 } },
+      )
+      .toArray();
+
+    const matchingPortalAmendments: PortalFacilityAmendment[] = [];
+
+    facilitiesOnDealWithPortalAmendments.forEach((facility) => {
+      if (facility.amendments) {
+        facility.amendments.forEach((amendment) => {
+          if (amendment?.type === AMENDMENT_TYPES.PORTAL && (!statuses || statuses.includes(amendment.status))) {
+            matchingPortalAmendments.push(amendment);
+          }
+        });
+      }
+    });
+
```

Comment:
Do you think this is clearer? (I havent tested it yet but I think this does the same)
```suggestion
    const matchingPortalAmendments = facilitiesOnDealWithPortalAmendments
      .flatMap((facility) => facility.amendments || [])
      .filter((amendment) => amendment?.type === AMENDMENT_TYPES.PORTAL && (!statuses || statuses.includes(amendment.status)));
```

Reply:
Much clearer, thank you!! Have changed